cmd.go

```diff
@@ -438,6 +439,14 @@ func syncList(ctx context.Context, run *runtime.Runtime,
 		}
 		defer resp.Body.Close()
 
+		if resp.Header.Get("Content-Encoding") == "" {
+			gzReader, err := gzip.NewReader(resp.Body)
+			if err == nil {
+				// Replace the raw binary body with the decompressing reader
+				resp.Body = gzReader
+			}
+		}
+
 		scanner := bufio.NewScanner(resp.Body)
```

Comment:
Good fix for handling the Content-Encoding header! A few technical issues to address:

  1. Error handling: Silently ignoring the error makes debugging harder. If decompression fails, we should know about it
  2. Resource management: gzip.Reader allocates internal buffers that need to be closed (per Go's compress/gzip docs), otherwise memory accumulates on repeated calls
  3. Type assignment: Assigning gzReader to resp.Body has type compatibility issues. Better to use a separate io.Reader variable


```suggestion
		var reader io.Reader = resp.Body
		if resp.Header.Get("Content-Encoding") == "" {
			gzipReader, err := gzip.NewReader(resp.Body)
			if err != nil {
				return fmt.Errorf("failed to decompress packages.gz: %w", err)
			}
			defer gzipReader.Close()
			reader = gzipReader
		}

		scanner := bufio.NewScanner(reader)
```

Reply:
Thanks @cockroach-eater! Go is not my native tongue ;)

The compiler didn't like the idea of assigning to the reader as you proposed. I've pushed a slightly different approach. Please take a look.

Reply:
Thanks @espeer!
It's not my native language either, suffering a bit, but I'm really motivated to fix this bug.
And also, thanks a lot for starting and maintaining this pr in the first place 🙌